src/udev/udev-rules.c

```diff
@@ -1313,10 +1321,29 @@ int udev_rules_load(UdevRules **ret_rules, ResolveNameTiming resolve_name_timing
 }
 
 bool udev_rules_check_timestamp(UdevRules *rules) {
```

Comment:
so i guess it is OK to make this change, but I think this is not as relevant as it might appear, since package managers and editors should change files atomically these days, i.e. create a new temporary inode, populate it, and then rename it over the old one. That way you'll always generate mtime updates on the parent dir...

Hence, in any realistic world I am pretty sure this should be a non-issue. But then again, checking this fully certainly shouldn't hurt, given that this is only done on request anyway, iiuc, i.e. not done frequently.

Reply:
> But then again, checking this fully certainly shouldn't hurt, given that this is only done on request anyway, iiuc, i.e. not done frequently.

This is called once per 3 sec, or when `udevadm control --reload` is invoked. Hence, this should not cause performance regression.